lib/Twig/Extension/Core.php

```diff
@@ -1507,12 +1507,15 @@ function twig_get_attribute(Twig_Environment $env, Twig_Source $source, $object,
                 continue;
             }
 
-            if (!isset($cache[$name])) {
-                $cache[$name] = $method;
-                $cache[$lcName] = $method;
+            if (!isset($classCache[$name])) {
+                $classCache[$name] = $method;
+            }
+
+            if (!isset($classCache[$lcName])) {
```

Comment:
I messed up when moving the method to a function as there is now a clash between a local `$cache` var and the global one. Your change fixes this in a good way, but we could have used the same var directly.